### ISSUE TITLE
Support HTTP/2

### ISSUE THREAD

**Issue by futurely** [ReactiveX/RxNetty#338]
HTTP/2 has been [finalized last month](https://www.google.co.in/?gws_rd=ssl#newwindow=1&q=http%2F2+finalized) and Netty has already implmented [the protocol](https://github.com/netty/netty/tree/master/codec-http2) and [an example](https://github.com/netty/netty/tree/master/example/src/main/java/io/netty/example/http2).

**Comment by NiteshKant** [comment]
@futurely Yes it will be provided in 0.5.x. 
Since, there are plenty of big changes in 0.5.x, I am currently [finishing those up](https://github.com/ReactiveX/RxNetty/milestones/0.5.0) before I can start implementing the HTTP/2 support.

Do you have a usecase for HTTP/2?

**Comment by futurely** [comment]
Yes, I want to exploit the server push capability of HTTP/2. WebSocket used to be the de facto choice to implement persistent connection based push notification services. For example, AeroGear implemtend Mozilla's [SimplePush protocol](https://wiki.mozilla.org/WebAPI/SimplePush/Protocol) with [WebSocket](https://github.com/aerogear/aerogear-simplepush-server). Recently, Mozilla has proposed a new protocol [webpush based on HTTP/2](https://datatracker.ietf.org/wg/webpush/documents/) and AeroGear [has implemented it](https://github.com/aerogear/aerogear-webpush-server) too. 

But it is still a proof of concept. In production, everything should be asynchronous as demonstrated by [Facebook's HBase RAFT quorum](https://github.com/apache/hbase/blob/HBASE-12259/hbase-consensus/src/main/java/org/apache/hadoop/hbase/consensus/client/QuorumThriftClientAgent.java). Without RxNetty, there would be a lot verbose SettableFutures and Runnables floating around and the logically related code has to be scattered. 

I would appreciate a lot if you can devote your precious time into this issue.

**Comment by NiteshKant** [comment]
Thanks @futurely for the context.

We at Netflix, are looking at HTTP/2 to be used as the default for our IPC stack and that is the reason for me to backport netty's HTTP/2 implementation from 5.x to 4.x. Netty 4.1 was recently released with HTTP/2.

We are totally invested in providing HTTP/2 support, I would expect it to be available sometime in April.

**Comment by futurely** [comment]
Thanks! I will experiment with RxNetty WebSocket based push servers before HTTP/2 is in place.

**Comment by schmitch** [comment]
Any Upgrades on this?

**Comment by futurely** [comment]
http://www.grpc.io/

**Comment by NiteshKant** [comment]
@schmitch we aren't working on HTTP/2 support as yet and I do not have any timelines on when we will work on it. 

@futurely Grpc AFAIK is a different interaction model and space than RxNetty but yeah if it works for your usecase, it is great!

**Comment by sundar-gs** [comment]
@NiteshKant  Do we have HTTP/2 support now in rxnetty?

**Comment by jchapuis** [comment]
Any progress on this? It would be great to have backpressure plugging into HTTP/2 flow control mechanisms transparently, e.g. exposing a Flowable to stream a HTTP/2 response. It seems that netty provides the low level plumbing to make this possible, is there any catch I don't see?

**Comment by moderakh** [comment]
@NiteshKant Our SDK has dependency rxNetty: 
https://mvnrepository.com/artifact/com.microsoft.azure/azure-cosmosdb/1.0.2 
https://github.com/Azure/azure-cosmosdb-java

If rxNetty adds support for Http/2 it will benefit us too.

Thanks.
